analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -0,0 +1,36 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks.CSharp
+{
+    public class NullPointerDereference : SymbolicRuleCheck
```

Comment:
Let's directly create a base class. We know that we will add VB version too. It will help with placing the correct code into a correct file (everything `ShouldExecute`-related here, everything rule-related to the SonarAnalyzer.Common)

Constant and the static field can already go to the base.

Reply:
> We know that we will add VB version too

Just to make sure: There isn't a VB version at this point. Do we really need to support VB in the future?

Reply:
Ah, the coverage bites :/ Let's remove it then (keep it somewhere so you can paste it when needed)